mergify_engine/tests/unit/test_json.py

```diff
@@ -0,0 +1,44 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2018–2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import enum
+
+from mergify_engine import json
+
+
+class Colour(enum.Enum):
+    RED = 1
+    GREEN = 2
+    BLUE = 3
+
+
+with_enum = {
+    "name": "hello",
+    "conditions": [],
+    "actions": {"merge": {"strict": Colour.BLUE}},
+}
+
+with_enum_encode = '{"name": "hello", "conditions": [], "actions": {"merge": {"strict": {"__pytype__": "enum", "class": "Colour", "name": "BLUE"}}}}'
+
+
+def test_encode_enum():
+    json.register_type(Colour)
+    assert json.dumps(with_enum) == with_enum_encode
```

Comment:
I think it's fine but be aware this can fail in theory.
A dict is not ordered, so the order in which the strings are printed in the string `with_enum_encode` might change. Which would break the test.

Reply:
if I use `assert data.get("actions").get("merge").get("strict") == Colour.BLUE` is it better ?

Reply:
what's `data` in your example? 🤔

Reply:
sorry :
```
data = json.loads(json_file)
assert data.get("actions").get("merge").get("strict") == Colour.BLUE
```

Reply:
or I can write it like this:

```
assert (
        json.loads(json_file).data.get("actions").get("merge").get("strict")
        == Colour.BLUE
    )
```

Reply:
If you use `mergify.json.loads` therefore your just writing the same test as `test_decode_enum`.

What you should rather do is use `json.loads` (not the Mergify one) and check the returned data, so:

```python
assert json.loads(json_file) == {"name": "hello", "conditions": [], "actions": {"merge": {"strict": {"__pytype__": "enum", "class": "Colour", "name": "BLUE"}}}}
```